src/sim_engine/filesystem.rs

```diff
@@ -25,4 +25,11 @@ impl Filesystem for SimFilesystem {
         };
         Box::new(filesystem_copy)
     }
+    fn get_id(&self) -> String {
+        self.mount_point.clone()
+    }
+
+    fn eq(&self, other: &Filesystem) -> bool {
+        self.get_id() == other.get_id()
```

Comment:
The other parameter in this case is a trait - Filesystem - it doesn't have a mountpoint like SimFilesystem.  I could add a get_mountpoint(), but it would have to do the same thing.